### ISSUE TITLE
[WiiU] Folders aren't being automatically created when needed.

### ISSUE THREAD

**Issue by Ploggy** [libretro/RetroArch#14487]
# First and foremost consider this:
- Only RetroArch bugs should be filed here. Not core bugs or game bugs
- This is not a forum or a help section, this is strictly developer oriented

## Description
On WiiU folders are not being created when needed, if I attempt to download thumbnails or save a remap file they wont be downloaded or made..

However IF I create the necessary folders manually before hand thumbnails/saving remaps files work as intended.

Example for Thumbnails:
Say I have a Snes playlist with games and I want to download a thumbnail I would select a game and select download thumbnail.. this will create a folder called "Nintendo - Super Nintendo Entertainment System" inside the "Thumbnails" folder, this folder does NOT get created with WiiU.

Example for Remaps:
Say I load a snes game, open the in game menu, edit the control config then Save Game Remap File, it will create a folder called "Snes9x" in Config/Remaps.. on WiiU the folder wont be created.

This bug probably effects more than just thumbnails or remaps but these are 2 examples of the issue.

A work around would be to create every needed folder before hand but that's not practical :P

### Expected behavior
Folders should automatically be created when thumbnails or remap files are made

### Actual behavior
Folders that are required for thumbnails and remap files aren't being created

### Steps to reproduce the bug

1. Boot retroarch
2. Attempt to download a thumbnail or save a remap file
3. 

### Bisect Results

from my understanding this has always happened.

### Version/Commit
You can find this information under Information/System Information

- RetroArch: [version/commit]

### Environment information

- OS: WiiU
- Compiler: [In case you are running local builds]

**Comment by LibretroAdmin** [comment]
Hi @Ploggy ,

can you confirm if this fixes the issue? Thanks.

https://github.com/libretro/RetroArch/pull/14490

**Comment by Ploggy** [comment]
I have tried with the new changes by gblues, but the folder creation issue is still there. Both saving a remap file still errors and thumbnail folders aren't being made when needed.

I also tried content scanning, it scans my 32X directory and says its completed successfully but there is no new 32X playlist added
As for downloading Overlays from the Online Updater, it downloads fine, but says extraction failed.

**Comment by LibretroAdmin** [comment]
Are you sure you have been testing the proper version? Did you check the Git hash in System Information?

**Comment by Ploggy** [comment]
This is the git hash from the core I built to test:
 
![20221007_202003](https://user-images.githubusercontent.com/1160255/194638453-d834497b-43b9-469a-bfd8-90a8a7ccc892.jpg)


I also grabbed an artifact from here https://git.libretro.com/libretro/picodrive/-/jobs/2595981 too which apparently was built after the pr was merged..

I'm still waiting in the usual buildbot to 100% confirm tho.

**Comment by Ploggy** [comment]
Confirmed, the issue is the same even with the new Buildbot build.

**Comment by Ploggy** [comment]
I've noticed the Thumbnail and Overlay path are not hard coded like the other paths they just say Default, actually now that I'm looking there are other paths that say default too..
Dynamic Backgrounds
Audio Filters
Video Shaders
Recordings
Controller Profiles
Favorites Playlist
Music Playlist
Video Playlist
Runtime Logs
All say "Default"
Cheat Files actually says "None" :P

Could they be hard coded to say:
sd:/retroarch/thumbnails
sd:/retroarch/overlays
Etc.. Etc..
or wouldn't that make any difference to the issue we have?

**Comment by gblues** [comment]
Oh! Yes, that actually is relevant.

Add these lines to your retroarch.cfg:

```
thumbnails_directory = "sd:/retroarch/thumbnails"
overlay_directory = "sd:/retroarch/overlay"
```
We probably should give these sensible defaults; "default" tends to translate to an empty string, at least for the thumbnails directory.

**Comment by LibretroAdmin** [comment]
@Ploggy Test if it works now - the PR by gblues has been merged which should create missing default directories.

**Comment by Ploggy** [comment]
> @Ploggy Test if it works now - the PR by gblues has been merged which should create missing default directories.

With that pr the new default directories are now set correctly. :)

Whats working:
Overlays.zip is downloading and extracting correctly!
Directories now have new set default paths!

Whats not working:
New folders aren't being created when needed, which means:
Thumbnails aren't being saved (due to needed folder not being created)
Remaps files aren't being saved (due to needed folder not being created)

There's no % during extraction process (not really an issue as such but it should be there?)

**Comment by LibretroAdmin** [comment]
So what you are telling me is that the mkdir issue is still there? Back to you I guess @gblues

**Comment by gblues** [comment]
It's working for me, it's not clear yet why it's not working for @Ploggy .

**Comment by Ploggy** [comment]
@LibretroAdmin we figured out why thumbnails were working for gblues but not for me, it was due to Playlists. I was using Manual Scan to create my playlist and gblues was using Scan Directory for his Playlists.
With Manual Scan the scan relies on correct romnames where as with Scan Directory the scan uses the roms CRC so even if the romname doesn't match the Thumbnail's name it will still download and show correctly.

Long story short if you use Manual Scan make sure your romsname matches the thumbnail name from the thumbnail site.
http://thumbnails.libretro.com/
